spec/puppet-server-lib/puppet/jvm/execution_spec.rb

```diff
@@ -22,6 +22,13 @@ def test_execute(command, options = {})
       expect(result).to eq "hi\n"
     end
 
+    it "should combine STDOUT/STDERR of the process when :combine is set" do
+      result = test_execute("./spec/fixtures/puppet-server-lib/puppet/jvm/execution_spec/echo_stdout_and_stderr.sh",
+                            {:combine => true})
+      expect(result).to match(/hello stdout/)
```

Comment:
It seems non-deterministic at the OS level, because it's about the timing of flushes, not just writes.